pkg/networkservice/metrics/server.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metrics - implement vpp based metrics collector service, it update connection on passing Request() with set of new metrics received during interval
+package metrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.ligato.io/vpp-agent/v3/proto/ligato/configurator"
+	vpp_interfaces "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/interfaces"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/vppagent"
+)
+
+// Server - helper interface to get some various information required for testing.
+type Server interface {
+	// GetMetricsReceived - show how many events we receive from internal vpp agent inerface
+	GetMetricsReceived() int
+	networkservice.NetworkServiceServer
+}
+
+type metricsServer struct {
+	interval        time.Duration
```

Comment:
Do we have options for polling stats that are not based on an interval timer?

Reply:
Nope unfortunately.

Reply:
I just talked to @ondrej-fabry he's been thinking of using a 0 poll interval for immediate response...

Reply:
Interesting... Will zero poll interval load CPU?

Reply:
That's a good point for discussion... I think 0 poll interval would simply return *once* immediately...

The really interesting question here is how to gather stats... we would like to reduce the number of timers in the system... and a Server level stats timer might be good for some Requestors, but not others... I'm quite open to discussion here.

Reply:
if vpp-agent already had stats for some interval, 0-interval request will be ideal. Thanks.